csharp/ql/src/Security Features/CWE-020/RuntimeChecksBypass.ql

```diff
@@ -25,11 +25,19 @@ Expr checkedWrite(Field f, Variable v, IfStmt check) {
   result.getAControlFlowNode() = check.getAControlFlowNode().getASuccessor*()
```

Comment:
Now that we are at it, I think this should be rewritten to use the guards library instead.

Reply:
I did this originally but then realised that the guards library does not (yet) support inequalities, which is what I used in the sample. Can we do this work another time?

Reply:
I think this would work (better):
```
GuardedExpr checkedWrite(Field f, Variable v, IfStmt check) {
  result = v.getAnAccess() and
  result = f.getAnAssignedValue() and
  check.getCondition().getAChildExpr*() = result.getAGuard()
}
```